### PR TITLE
docs: Improve order, add FAQ and welcome page

### PR DESCRIPTION
**Pull request by lukeblevins**
GitHub Issue (If applicable): N/A

## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/ExtensionsOverview.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Uno.Extensions is a series of NuGet packages designed to encapsulate common functions when building multi-platform mobile, desktop and web applications using Uno Platform. Thoughtfully designed components and features retain easy extensibility and customization, while providing a solid foundation for your application.
+
+[placeholderimagehere]()
+
+The Uno.Extensions follows the Microsoft.Extensions model that creates a host environment where you can register additional dependencies. The registered dependencies are then available throughout the application via Services (`IServiceProvider`) property on the `IHost` instance.
+
+## Learn about Uno.Extensions features
+
+Alongside Uno, the app template references a lineup of battle-tested packages providing key features. The resources below are available to help you get started with them in your project.
```
Comment (agneszitte-nventive) [inline doc/ExtensionsOverview.md:14]:
```suggestion
# Get to a fully functional project in minutes

Uno.Extensions is a series of NuGet packages designed to encapsulate common functions when building multi-platform mobile, desktop, and web applications using Uno Platform. Thoughtfully designed components and features retain easy extensibility and customization while providing a solid foundation for your application.

[placeholderimagehere]()

The Uno.Extensions follows the Microsoft.Extensions model that creates a host environment where you can register additional dependencies. The registered dependencies are then available throughout the application via the Services (`IServiceProvider`) property on the `IHost` instance.

## Learn about Uno.Extensions features

Alongside Uno, the app template references a lineup of battle-tested packages providing key features. The resources below are available to help you get started with them in your project.
```

File: doc/GettingStarted.md [hunk 2]
```diff
@@ -1,31 +1,9 @@
...
-| Uno.Extensions.Logging       | Configures application logging                                                       |
-| Uno.Extensions.Navigation    | Routing framework for navigation                                                     |
-| Uno.Extensions.Reactive      | Development framework for reactive applications                                      |
-| Uno.Extensions.Localization  | Provides access to localizable resources                                             |
-| Uno.Extensions.Serialization | Helpers for json serialization                                                       |
-| Uno.Extensions.Http          | Configures native http handlers and register endpoints                               |
-
-
-This tutorial will walk through how to create an Uno application with the `dotnet new` tool, that is already configured to use the Uno.Extensions.
+This tutorial will walk through how to create an Uno application with the `dotnet new` tool, that is already configured to use the Uno Extensions.
```
Comment (agneszitte-nventive) [inline doc/GettingStarted.md:6]:
```suggestion
This tutorial will walk through how to create an Uno application with the `dotnet new` tool, which is already configured to use the Uno Extensions.
```

File: doc/Questions.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Likewise, your organization can streamline developer resources by reusing tested code across both iOS and Android devices.
+
+## How does Uno Extensions allow me to extend the functionality of my app?
+A host environment is provided where you can easily register application features as additional dependencies. The registered dependencies are then available throughout the application.
+
+## How can I get started with Uno Extensions?
+Instructions for creating an application from the template can be found [here](xref:GettingStarted)
+
+## What are the NuGet packages included in Uno Extensions?
+The Uno Extensions collection includes the following NuGet packages:
```
Comment (agneszitte-nventive) [inline doc/Questions.md:29]:
```suggestion
# Frequently Asked Questions

## What is Uno Extensions?
Uno Extensions is our lineup of helpful NuGet packages designed to encapsulate common functions when building multi-platform mobile, desktop, and web applications with Uno Platform.

## What functionality is part of the Uno Extensions collection?
Key features like dependency injection, serialization, configuration, environments, HTTP, localization, logging, hosting, and authentication allow you to save time while also lowering your overall budget for the application. 

## How productive will I be with Uno Extensions?
With Uno Extensions, you'll be bursting with productivity! 

From advancements in starting a new project with the app template to the ability to extend the functionality of your app, you'll be able to get to market faster and with less effort.

## How can Uno Extensions help me meet the moment?
In a world of heightened macroeconomic volatility, you need to be able to respond quickly to changing market conditions. Uno Extensions allows you to do just that by saving up months of development time. You will be able to reduce the budget needed to develop the app and reap the benefits of arriving to market fast.

Likewise, your organization can streamline developer resources by reusing tested code across both iOS and Android devices.

## How does Uno Extensions allow me to extend the functionality of my app?
A host environment is provided where you can easily register application features as additional dependencies. The registered dependencies are then available throughout the application.

## How can I get started with Uno Extensions?
Instructions for creating an application from the template can be found [here](xref:GettingStarted)

## What are the NuGet packages included in Uno Extensions?
The Uno Extensions collection includes the following NuGet packages:
```

File: doc/Questions.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+| Package                      |                                                                                      |
+|------------------------------|--------------------------------------------------------------------------------------|
+| Uno.Extensions.Hosting       | Initializes the hosting environment, including initialising the dependency container |
```
Comment (agneszitte-nventive) [inline doc/Questions.md:33]:
```suggestion
| Uno.Extensions.Hosting       | Initializes the hosting environment, including initializing the dependency container |
```

File: doc/GettingStarted.md [hunk 5]
```diff
@@ -1,31 +1,9 @@
 ---
-uid: Overview.Extensions
+uid: GettingStarted
```
Comment (nickrandolph) [inline doc/GettingStarted.md:2]: @lukeblevins were there any references to the Overview.Extensions id? If so, they need to be updated

Comment (lukeblevins) [reply doc/GettingStarted.md]: Handled in 1434dca3da8a42881f1158bc2e5edb5e87b94d9f

File: doc/toc.yml [hunk 6]
```diff
@@ -1,115 +1,118 @@
...
-  href: Overview/ExtensionsOverviewAndGettingStarted.md
+  href: ExtensionsOverview.md
+- name: Frequently Asked Questions
+  href: Questions.md
+- name: Creating an Application
+  href: GettingStarted.md
+# - name: Overview topics (legacy)
+#   items:
+#     - name: Hosting
+#       href: Overview/Hosting/HostingOverview.md
```
Comment (nickrandolph) [inline doc/toc.yml:10]: @lukeblevins where are all the overall topics now? I don't see them anywhere in the TOC?

Comment (lukeblevins) [reply doc/toc.yml:10]: I thought that we decided to remove that section because it's incomplete?

Comment (lukeblevins) [reply doc/toc.yml:10]: My bad, I will add that part back.

Comment (lukeblevins) [reply doc/toc.yml]: I've integrated these topics back in a way which feels more intuitive and mirrors the general structure used by Microsoft. We now consolidate Learn and Overview into "Features"

**Comment by lukeblevins** [comment]
I've validated these changes work properly in the local environment. Here's a visual comparison (old: left; new: right)
image

File: doc/toc.yml [hunk 7]
```diff
@@ -1,19 +1,89 @@
 - name: Overview
-  href: Overview/ExtensionsOverviewAndGettingStarted.md
+  href: ExtensionsOverview.md
+- name: Frequently Asked Questions
+  href: Questions.md
+- name: Creating an Application
+  href: GettingStarted.md
+- name: Features
   items:
+    - name: Authentication
```
Comment (nickrandolph) [inline doc/toc.yml:9]: These should be in alphabetic order - eg Hosting isn't correct

**Comment by nickrandolph** [comment]
@francoistanguay are you ok with the proposed changes to the hierarchy?

File: doc/toc.yml (hunk 7 above)
Comment (lukeblevins) [reply doc/toc.yml:9]: @nickrandolph corrected in https://github.com/unoplatform/uno.extensions/pull/1089/commits/c307bf604dd56a54cae3d7177cf41713477f5585

**Comment by sasakrsmanovic** [comment]
A few things on my side - first on semantics


1. Reduce marketing talk: i.e. this section: How can Uno Extensions help me meet the moment?
In a world of heightened macroeconomic volatility, you need to be able to respond quickly to changing market conditions. Uno Extensions allows you to do just that by saving up months of development time. You will be able to reduce the budget needed to develop the app and reap the benefits of arriving to market fast.

Likewise, your organization can streamline developer resources by reusing tested code across both iOS and Android devices.


2. For brand and repo consistency I think we should use Uno Platform Extensions, often simply referred to as "Uno Extensions" in the first paragraph; then it is OK to call it Uno Extensions after



-

**Comment by sasakrsmanovic** [comment]
The second, structural, issue is with the way the menu changed as all other sections have the same menu flow, with some [variations] 

1. Overview
2. [Learn]
3. [Development]
4. Reference

What I like about Luke's PR is that it brings out Overview right up. But it also seems to 'hide' Learn part. 

I think it is beneficial to have similar sub-sections in each section for docs readability. If we decide to do this in with Extensions we should probably do the same to other sections too

**Comment by lukeblevins** [comment]
@sasakrsmanovic Thanks for the great feedback!

Regarding your second point, would it make sense to change the "Features" node to say "Learn"? That way, users are clear that tutorials exist under the sections. My intention was to consolidate those sections because "Overview" should exist as a first step before diving into tutorials.

**Pull request merged**